cmd/snap-confine/snap-confine.c

```diff
@@ -414,12 +422,37 @@ int main(int argc, char **argv) {
 
     if (cap_set_flag(caps_no_effective, CAP_PERMITTED, SC_ARRAY_SIZE(keep_no_effective_caps), keep_no_effective_caps,
                      CAP_SET) != 0) {
-        die("cannot set capapbility flags");
+        die("cannot set capability flags");
+    }
+
+    if (sc_cgroup_is_v2()) {
+        /* When we are using cgroup v2 then we don't need to use SETUID and
+         * SETGID capabilities and we can drop them from the permitted set.
+         * When we are using cgroup v1 then we actually need them to manipulate
+         * the freezer, so set them in the effective set. We drop the
```

Comment:
"...so we set them", otherwise it sounds like an imperative but actually the stanza does the reverse here and is dropping stuff

Reply:
I rewrote that part.